docs/ReleaseProcess.md

```diff
@@ -25,41 +27,69 @@ eBPF for Windows, and how to service it later.
    (accessible via the "`Actions`" tab on GitHub).
-1. Extract the `*.nupkg` file, and rename it in the following format (replace "`X.Y.Z`" with the version number being released):
+7. Extract the `*.nupkg` file, and rename it in the following format:
+    - `eBPF-for-Windows.X.Y.0.nupkg`
+8. Mark the draft PR as "Ready for review".
+9. Once the PR is approved and merged into the "`main`" branch, of the upstream `ebpf-for-windows` repo, create a new release branch from `main` from the **previous PR's commit**, and name it "`release/X.Y.`".
+10. **IMPORTANT:** Once the release branch is created, no new feature work is allowed to be merged. However, bug fixes can be taken as deemed necessary by the maintainers and the release manager. Whenever applicable, these bug fixer should first be made in the main branch, and cherry-picked to the release branch. In case a bug-fix PRs are merged into the `release/X.Y` branch, the **latest commit** in the `release/X.Y` branch must be designated as the commit for the release.
+11. Follow the process in the [Release Branch Validation](ReleaseProcess.md#release-branch-validation) to ensure the quality of the release branch.
+1. In the triage meeting after successful creation of the release branch, the release manager must ask if any maintainer has any reasons to hold off the release. If not, move to the following steps.
+4. The `sign-off` label will be added to the issue created in step 1 of the [Creating a new release](ReleaseProcess.md#creating-a-new-release) process. Next create a tag for the *latest commit on the `release/X.Y` branch*. The tag should reflect the version number being released and adhere to the following notation: "`Release-vX.Y.0`".
+5. The tag creation will automatically trigger the "`CI/CD - Release validation`" workflow for the `release/X.Y` branch. In case of failure, Follow the process in the [Release Branch Validation](ReleaseProcess.md#release-branch-validation).
+13. Publish the release as per the [Publishing a Release](ReleaseProcess.md#publishing-a-release) process.
+
+## Release Branch Validation
+
+The `CI/CD - Release validation` worfklow (`cicd-release-validation.yml`) is used to validate a release branch. It contains more tests than the regular CI/CD pipeline, including longer duration fuzz tests, fault injection tests, stress tests, performance tests etc. These tests can be manually scheduled. Release manager must run these tests on the release manager. Due to the non-deterministic nature of some of the tests, it is recommended that the tests are run at least three times on the branch. If any of the tests fail, the release manager must investigate the failure and follow up with issues in github. Once potential fixes are merged to the release branch repeat the process until the workflow completes successfully.
```

Comment:
IMHO, you should never find out about a release problem after you fork the release. If you have some tests you don't/can't run on a PR, fine. But you should be running on commits to main, **before** you release.

Reply:
The release validation should run on a release branch where there will not be any churns. We do not want to block new commits to the main branch after forking the release branch.

Reply:
My argument here is that the historical 'Windows process' of fork and then figure out your bugs and fix them shouldn't be the model to follow. Windows has to do that because of its size, but not undocked projects. Ideally, projects should shoot for having **all** the necessary automation run on PRs to keep your product clean (i.e. never merge a bug). But that often requires too much time, and therefore we relax some of the requirements in effort to speed up the inner dev loop. But that validation should then run on official commits immediately after merge. This way, you don't block progress, but you **immediately** find out the problem. I will admit this is a goal, and not necessarily something easily achievable, but it should be what we all shoot for; not the Windows process.